src/software/thunderscope/play/refereeinfo_widget.py

```diff
@@ -87,17 +84,17 @@ def refresh(self) -> None:
 
         for info in team_info:
             if info == "yellowCardTimes":
-                blue.append(self.parse_yellow_card_times(referee_msg_dict["blue"]))
-                yellow.append(self.parse_yellow_card_times(referee_msg_dict["yellow"]))
+                blue.append(self.parse_yellow_card_times(referee.blue()))
+                yellow.append(self.parse_yellow_card_times(referee.yellow()))
             elif info == "remainingTimeouts":
-                blue.append(referee_msg_dict["blue"]["timeouts"])
-                yellow.append(referee_msg_dict["yellow"]["timeouts"])
+                blue.append(referee.blue().timeouts())
+                yellow.append(referee.yellow().timeouts())
             elif info == "goalkeeperID":
-                blue.append(referee_msg_dict["blue"]["goalkeeper"])
-                yellow.append(referee_msg_dict["yellow"]["goalkeeper"])
+                blue.append(referee.blue().goalkeepers())
+                yellow.append(referee.yellow().goalkeeper())
             else:
-                blue.append(referee_msg_dict["blue"][info])
-                yellow.append(referee_msg_dict["yellow"][info])
+                blue.append(referee.blue().info())
+                yellow.append(referee.yellow().info())
 
         set_table_data(
             {
```

Comment:
you also need to update the implementation of `parse_yellow_card_times` I believe for the team_info object